src/main/java/org/testeditor/fixture/web/json/BrowserSettingsManager.java

```diff
@@ -40,36 +43,153 @@
+        List<BrowserSetupElement> elements = reader.readElements(fileName);
+        Platform currentPlattform = Platform.getCurrent();
+        return getBrowserElementsWithoutDuplicateEntries(elements,currentPlattform);
+    }
     
-    private static List<BrowserSetupElement> getBrowserSpecificSetting(List<BrowserSetupElement> 
-        elements,Platform platform) {
-        List<BrowserSetupElement> settings = new ArrayList<>();    
-        settings = separateBrowserSettingsForOs(elements, platform);
-        separateBrowserSettingsOsUnspecific(settings, elements);
-        return settings;
+    protected static List<BrowserSetupElement> getBrowserElementsWithoutDuplicateEntries(
+            List<BrowserSetupElement> elements,Platform platform) {
+        List<BrowserSetupElement> separatedBrowserSettingsForOs = separateBrowserSettingsForOs(elements, platform);
+        List<BrowserSetupElement> separateBrowserSettingsOsUnspecific = separateBrowserSettingsOsUnspecific(elements);
+        Map<String,Object> capabilitiesOsUnspecific = copyCapabilitiesIntoMap(new HashMap(), 
+                separateBrowserSettingsOsUnspecific); 
+        copyCapabilitiesIntoMap(capabilitiesOsUnspecific,separatedBrowserSettingsForOs); 
+        Map<String,Object> optionsOsUnspecific = copyOptionsIntoMap(new HashMap(), 
+                separateBrowserSettingsOsUnspecific);  
+        copyOptionsIntoMap(optionsOsUnspecific,separatedBrowserSettingsForOs); 
+        separatedBrowserSettingsForOs.addAll(separateBrowserSettingsOsUnspecific);
+        return separatedBrowserSettingsForOs;
     }
-
-    private static void separateBrowserSettingsOsUnspecific(List<BrowserSetupElement> settings,
-            List<BrowserSetupElement> elements) {
+    
+    // just not deleted for your opinion. One method instead of 2 for options and capabilities
+    //    protected static Map<String,Object> separateBrowserSettings(List<BrowserSetupElement> 
+    //        separateBrowserSettingsOsUnspecific, String browserSettingType) {
+    //        Map<String,Object> browserSettings = new HashMap();
+    //        separateBrowserSettingsOsUnspecific.forEach((bsetting) -> { 
+    //            if (browserSettingType.equals(OPTIONS)) {
+    //                bsetting.getOptions().forEach((option) ->  { 
+    //                    browserSettings.put(option.getKey(), option.getValue());
+    //                } 
+    //                );
+    //            
+    //            } else if (browserSettingType.equals(CAPABILITIES)) {
+    //                bsetting.getCapabilities().forEach((capability) ->  { 
+    //                    browserSettings.put(capability.getKey(), capability.getValue());
+    //                    } 
+    //                );
+    //            }  
+    //        
+    //        }   
+    //        ); 
+    //        return browserSettings;
+    //    }
+    
+    protected static Map<String,Object> copyOptionsIntoMap(Map<String,Object> osUnspecificSettings,
```

Comment:
please rename osUnspecificSettings  (to e.g. 'browserSettingsMap') and rename separateBrowserSettingsOsspecific (to e.g. 'settings')